RFS/src/main/java/com/rfs/common/DeletingSourceRepoAccessor.java

```diff
@@ -0,0 +1,46 @@
+package com.rfs.common;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+
+/*
+ * Provides access to the underlying files in the source repo and deletes the files after the Stream is closed.  This
+ * is useful/interesting in the case where the files are large/numerous and you can easily re-acquire them - such as
+ * if they are being loaded from S3.
+ */
+public class DeletingSourceRepoAccessor extends SourceRepoAccessor {
+    public DeletingSourceRepoAccessor(SourceRepo repo) {
+        super(repo);
+    }
+
+    @Override
+    protected InputStream load(Path path) {
+        try {
+            return new DeletingFileInputStream(path);
+        } catch (Exception e) {
+            throw new CouldNotLoadRepoFile("Could not load file: " + path, e);
+        }
+    }
+
+    public static class DeletingFileInputStream extends FileInputStream {
```

Comment:
Probably easier to talk this one out.

Reply:
After discussion, we decided to keep this around in case we need it in the future but use the Default (non-deleting) version in RFS for now.  If the disk fills up from our s3 downloads, it should kill the process and naturally free up disk space by getting a new container.